src/main/flight/pid.c

```diff
@@ -167,30 +168,32 @@ void pidStabilisationState(pidStabilisationState_e pidControllerState)
 
 const angle_index_t rcAliasToAngleIndexMap[] = { AI_ROLL, AI_PITCH };
 
-static FAST_RAM filterApplyFnPtr dtermNotchFilterApplyFn;
-static FAST_RAM void *dtermFilterNotch[2];
-static FAST_RAM filterApplyFnPtr dtermLpfApplyFn;
-static FAST_RAM void *dtermFilterLpf[2];
-static FAST_RAM filterApplyFnPtr ptermYawFilterApplyFn;
-static FAST_RAM void *ptermYawFilter;
+static FAST_RAM filterApplyFnPtr dtermNotchApplyFn;
+static FAST_RAM void *dtermNotch[2];
```

Comment:
These pointers are not necessary - simply use corresponding filter state union

Reply:
@Fujin - Thanks!  The approach using those pointers is not my code, I didn't change anything there, just used it exactly as for the other filters on D.

If you could explain exactly how this code could be improved, that would be great.

Reply:
@ctzsnooze he means make a typedef union struct which has at least a dtermNotch[2] for holding the filter's state. then use that directly (instead if indirection).

you can check the dual stage gyro PR for how we did it there.

Reply:
I would only stuff it up if I tried to do that  :-)   

Either... I to keep this PR as is to just provide the functionality, and someone who knows better how to improve the code makes a separate PR to optimise it, or:

Someone more expert than I spells out exactly what needs to be done, i.e. gives me the exact code changes needed, and I will add them now, get it all done and dusted.

Reply:
Changing the line in question to `static FAST_RAM biquadFilter_t dtermNotch[2]` and removing `biquadFilter_t biquadFilterNotch[2]` and the indirection in the calling code should do the trick.

Reply:
Can't get these changes to compile...
line 171 'static FAST_RAM filterApplyFnPtr dtermNotchApplyFn;' replaced with 'static FAST_RAM biquadFilter_t dtermNotch[2]'
line 172      deleted 'static FAST_RAM void *dtermNotch[2];'
then....
line 215      'static biquadFilter_t biquadFilterNotch[2];' deleted 
don't know how to fix the 'indirection in the calling code'... assume it needs something done in 
line 216    'dtermNotchApplyFn = (filterApplyFnPtr)biquadFilterApply;'  
doesn't compile, can't figure why.  Sorry.